ipa-core/src/helpers/transport/in_memory/transport.rs

```diff
@@ -73,6 +73,7 @@ pub enum Error<I> {
 /// incoming messages.
 pub struct InMemoryTransport<I> {
     identity: I,
+    all_identities: Vec<I>,
```

Comment:
you don't need to pass that, it can be derived

```rust
    pub fn peer_count(&self) -> u32 {
        u32::try_from(self.connections.len()).expect("Less than 4B peers")
    }
```

Reply:
I wanted the API to return an iterator over identities `I` (instead of just integers) so that later I could use it directly in the `broadcast` function. In your example `peer_count` works fine for the generic `InMemoryTransport<I>` because it's just an `u32`, but since I was working with `I` I needed a generic way that works for any `I`. I could't derive a `ShardIndex` or a `HelperIdentity` on the same function.

It looks kind of redundant to pass a parameter that's derivable, but the problem is that I can only set it at a point where I know if it's a Helper or Shard `InMemoryTransport`.

One thing to consider is that this issue only happens in the `InMemoryTransport` which is shared for both Helper and Shard.

I could alternatively have `peer_count` and try to derive the peers using integers. Let me know.

Reply:
You may also be able to get it from the keys of the `connections` `HashMap` (plus `identity` for the local one).